code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -197,13 +220,15 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 	amount = min(amount, sum) //Can not take more air than tile has!
 	if(amount <= 0)
 		return null
+	var/ratio = amount / sum
+	
 	var/datum/gas_mixture/removed = new type
 	var/list/removed_gases = removed.gases //accessing datum vars is slower than proc vars
 
 	removed.temperature = temperature
 	for(var/id in cached_gases)
 		ADD_GAS(id, removed.gases)
-		removed_gases[id][MOLES] = QUANTIZE((cached_gases[id][MOLES] / sum) * amount)
+		removed_gases[id][MOLES] = QUANTIZE_TO_MOVE((cached_gases[id][MOLES] * ratio)

```

Comment:
What happens if you try and move something that has a molar count that's less then this? You'll just end up creating gas from nowhere right?

Reply:
yes, microquantities of gas, when gas near MINIMUM_MOLE_MANIPULATION_COUNT		0.001. 

In normal way it must be not called for zero mole gas, 
but it called now somewhere.

Reply:
I do not think allowing for gas to appear from nowhere is a good idea, it will either break a system or be abused by players.

Reply:
this creating is unseable. If it noticeably something going wrong. Gas dont created from nowhere,
 to create air it must be at lower than mininun 0.001 moles and try take part of it.

Reply:
if we try take 50% of 0.001 moles of gas we move 0.001 moles. 
as i think if we have mole count lower than minimum outside of process (just lay on turf or in pipe or in ), something going wrong. If we have 0.003 moles and take 50% we take 0.0015 moles but QUANTIZE must change it to 0.002 moles, moles that we actual move.

Reply:
You madman. Maybe we should change it to an if. So if it's above the value we return it, and if not we return 0. I wonder....

Reply:
I'ma try that tomorrow. I wonder.....

Reply:
Need test that `round(variable,0.001)` round 0.0005 to 0.001 and 0.0004 to 0